src/term/mod.rs

```diff
@@ -436,18 +438,70 @@ impl Term {
             size_info: size,
             empty_cell: template,
             custom_cursor_colors: config.custom_cursor_colors(),
+            semantic_escape_chars: config.semantic_escape_chars().clone(),
         }
     }
 
     pub fn update_config(&mut self, config: &Config) {
-        self.custom_cursor_colors = config.custom_cursor_colors()
+        self.custom_cursor_colors = config.custom_cursor_colors();
+        self.semantic_escape_chars = config.semantic_escape_chars().clone();
     }
 
     #[inline]
     pub fn needs_draw(&self) -> bool {
         self.dirty
     }
 
+    pub fn line_selection(&self, selection: &mut Selection, point: Point) {
+        selection.clear();
+        selection.update(Point {
+            line: point.line,
+            col: Column(0),
+        }, Side::Left);
+        selection.update(Point {
+            line: point.line,
+            col: self.grid.num_cols() - Column(1),
+        }, Side::Right);
+    }
+
+    pub fn semantic_selection(&self, selection: &mut Selection, point: Point) {
+        let mut side_left = Point {
+            line: point.line,
+            col: point.col
+        };
+        let mut side_right = Point {
+            line: point.line,
+            col: point.col
+        };
+
+        let mut left_iter = self.grid.iter_from(point);
+        let mut right_iter = self.grid.iter_from(point);
+
+        while let Some(cell) = left_iter.prev() {
+            if self.semantic_escape_chars.contains(cell.c) {
+                break;
+            }
+            side_left.col = left_iter.cur.col;
+            side_left.line = left_iter.cur.line;
+
+            if left_iter.cur.col == Column(0) {
+                break; // cut off at beginning of line
```

Comment:
I did some testing in iTerm, and it looks like double click will actually work across lines *if* it detects that the line has wrapped. You can test this by echoing a URL that spans lines.

To accomplish the same thing in Alacritty, we would decide to continue expanding the selection when `iter.cur.col == grid.num_cols() - 1` and `cell.flags.contains(flags::WRAPLINE)`.

This case is distinct from when there is a linefeed (eg `\n`) which should break the selection. I think that behavior should fall out of the check described above.